tools/syz-fillreports/fillreports.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+// syz-fillreports queries all open bugs from a namespace, extracts the missing reporting elements
+// (currently only missing guilty files are supported) and uploads them back to the dashboard.
+
+package main
+
+import (
+	"flag"
+	"log"
+	"sync"
+
+	"github.com/google/syzkaller/dashboard/dashapi"
+	"github.com/google/syzkaller/pkg/mgrconfig"
+	"github.com/google/syzkaller/pkg/report"
+	_ "github.com/google/syzkaller/sys"
+	"github.com/google/syzkaller/sys/targets"
+)
+
+var (
+	flagDashboard = flag.String("dashboard", "https://syzkaller.appspot.com", "dashboard address")
+	flagAPIClient = flag.String("client", "", "the name of the API client")
+	flagAPIKey    = flag.String("key", "", "api key")
+)
+
+func main() {
+	flag.Parse()
+
+	dash, err := dashapi.New(*flagAPIClient, *flagDashboard, *flagAPIKey)
+	if err != nil {
+		log.Fatalf("dashapi failed: %v", err)
+	}
+	resp, err := dash.BugList()
+	if err != nil {
+		log.Fatalf("bug list query failed: %v", err)
+	}
+	workItems := loadBugReports(dash, resp.List)
+	for item := range workItems {
+		processReport(dash, item)
+	}
+}
+
+func processReport(dash *dashapi.Dashboard, bugReport *dashapi.BugReport) {
+	if bugReport.ReportElements != nil && len(bugReport.ReportElements.GuiltyFiles) > 0 {
+		log.Printf("%v: already has guilty files", bugReport.ID)
+		return
+	}
+	if bugReport.BugStatus != dashapi.BugStatusOpen {
+		log.Printf("%v: status != BugStatusOpen", bugReport.ID)
+		return
+	}
+	cfg := &mgrconfig.Config{
+		Derived: mgrconfig.Derived{
+			TargetOS:   bugReport.OS,
+			TargetArch: bugReport.Arch,
+			SysTarget:  targets.Get(bugReport.OS, bugReport.Arch),
+		},
+	}
+	reporter, err := report.NewReporter(cfg)
+	if err != nil {
+		log.Fatalf("%v: failed to create a reporter for %s/%s",
+			bugReport.ID, bugReport.OS, bugReport.Arch)
+	}
+	rep := reporter.Parse(bugReport.Log)
+	if rep == nil {
+		log.Printf("%v: no crash is detected", bugReport.ID)
+		return
+	}
+	rep.Report = bugReport.Report
```

Comment:
Add a comment. It is tricky what happens here and relies on internal details of the current Symbolize implementation.